library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -2273,6 +2274,411 @@ void libspdm_set_last_spdm_error_struct(void *spdm_context, libspdm_error_struct
+{
+    libspdm_fips_selftest_context *libspdm_fips_selftest_context;
+    libspdm_context_t *libspdm_context;
+
+    libspdm_context = spdm_context;
+    libspdm_fips_selftest_context = fips_selftest_context;
+
+    if ((libspdm_context == NULL) || (libspdm_fips_selftest_context == NULL)) {
+        return false;
+    }
+    if (fips_selftest_context_size != sizeof(libspdm_fips_selftest_context)) {
+        return false;
+    }
+
+    libspdm_copy_mem(&(libspdm_context->fips_selftest_context),
+                     sizeof(libspdm_fips_selftest_context),
+                     libspdm_fips_selftest_context, sizeof(libspdm_fips_selftest_context));
+    return true;
+}
+
+/**
+ * export fips_selftest_context from spdm_context;
+ *
+ * @param[in]          spdm_context                A pointer to the spdm_context.
+ * @param[in,out]      fips_selftest_context       A pointer to the fips_selftest_context.
+ * @param[in]          fips_selftest_context_size  The size of fips_selftest_context.
+ *
+ * @retval true   export fips_selftest_context successful.
+ * @retval false  spdm_context or fips_selftest_context is null.
+ */
+bool libspdm_export_fips_selftest_context_from_spdm_context(void *spdm_context,
+                                                            void *fips_selftest_context,
+                                                            size_t fips_selftest_context_size)
+{
+    libspdm_fips_selftest_context *libspdm_fips_selftest_context;
+    libspdm_context_t *libspdm_context;
+
+    libspdm_context = spdm_context;
+    libspdm_fips_selftest_context = fips_selftest_context;
+
+    if ((libspdm_context == NULL) || (libspdm_fips_selftest_context == NULL)) {
+        return false;
+    }
+    if (fips_selftest_context_size != sizeof(libspdm_fips_selftest_context)) {
+        return false;
+    }
+
+    libspdm_copy_mem(libspdm_fips_selftest_context,
+                     sizeof(libspdm_fips_selftest_context),
+                     &(libspdm_context->fips_selftest_context),
+                     sizeof(libspdm_fips_selftest_context));
+    return true;
+}
+
+/**
+ * update fips_selftest_context in spdm_context;
+ *
+ * @param[in]    spdm_context                A pointer to the spdm_context.
+ *
+ * @retval true   run FIPS self_test successfully;
+ * @retval false  run FIPS self_test failed;
+ */
+bool libspdm_update_fips_selftest_context(void *spdm_context)
+{
+    libspdm_context_t *context;
+    bool result;
+
+    result = true;
+    context = spdm_context;
+    if (context == NULL) {
+        result = false;
+    }
+
+#if LIBSPDM_FIPS_TEST_AT_LOAD
+    if (context->fips_selftest_context.tested_algo == 0) {
+        if (!libspdm_fips_run_selftest()) {
+            result = false;
+        } else {
+            /*all algos are tested and the all results passed*/
+            context->fips_selftest_context.tested_algo =
+                LIBSPDM_FIPS_SELF_TEST_HMAC_SHA256 | LIBSPDM_FIPS_SELF_TEST_HMAC_SHA384 |
```

Comment:
It'd need #if LIBSPDM_XYZ_SUPPORT qualifier before participating in the | operation.